### PR TITLE
Feature/java doc updates

### PR DESCRIPTION
**Pull request by aebadirad**
Java doccing done to our public interface APIs

### CODE REVIEW

**Comment by aebadirad** [comment]
Yes, denoting what data hub operations are logically encapsulated in the interface offering.

File: marklogic-data-hub/src/main/java/com/marklogic/hub/FlowManager.java [hunk 1]
```diff
@@ -47,32 +51,52 @@ static Flow flowFromXml(Element doc) {
...
+    /**
+     * retrieves a list of all the flows on the local files systems
+     * @param entityName - string name of the entity for the flow
+     * @param flowType - the FlowType enum, eg: ingest or harmonize
+     * @return a list of Flows
+     */
     List<Flow> getLocalFlowsForEntity(String entityName, FlowType flowType);
 
+    /**
+     * Obtains a flow form a property file
```
Comment (paxtonhare) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/FlowManager.java]: form => from

File: marklogic-data-hub/src/main/java/com/marklogic/hub/flow/Flow.java [hunk 2]
```diff
@@ -21,26 +21,106 @@
...
     CodeFormat getCodeFormat();
 
+    /**
+     * Serializes the flow into an xml string
+     * @return a serialized xml string of the flow
+     */
     String serialize();
 
+    /**
+     *
```
Comment (paxtonhare) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/flow/Flow.java]: Creates a Properties object representing the flow

File: marklogic-data-hub/src/main/java/com/marklogic/hub/flow/FlowRunner.java [hunk 3]
```diff
@@ -23,18 +23,81 @@
...
+
+    /**
+     * Sets the source client on the flow runner
+     * @param sourceClient
+     * @return  the flow runner object
+     */
     FlowRunner withSourceClient(DatabaseClient sourceClient);
+
+    /**
+     * Sets which database the flow will output to
```
Comment (paxtonhare) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/flow/FlowRunner.java]: Sets the database where flow output data will be persisted

File: marklogic-data-hub/src/main/java/com/marklogic/hub/flow/FlowRunner.java [hunk 4]
```diff
@@ -23,18 +23,81 @@
...
+
+    /**
+     * Sets the thread count for the flowrunner
+     * @param threadCount - the number of threads for the flow runner to use
+     * @return the flow runner object
+     */
     FlowRunner withThreadCount(int threadCount);
+
+    /**
+     * Sets the source client on the flow runner
```
Comment (paxtonhare) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/flow/FlowRunner.java]: Sets the source client on the flow runner. The source client determines which database to run against for building the envelope.

File: marklogic-data-hub/src/main/java/com/marklogic/hub/flow/FlowRunner.java [hunk 5]
```diff
@@ -23,18 +23,81 @@
...
+
+    /**
+     * Sets which database the flow will output to
+     * @param destinationDatabase
+     * @return the flow runner object
+     */
     FlowRunner withDestinationDatabase(String destinationDatabase);
+
+    /**
+     * Sets the options for the flowRunner
```
Comment (paxtonhare) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/flow/FlowRunner.java]: Sets the options to be passed into the xqy or sjs flow in the $options or options variables of main.

File: marklogic-data-hub/src/main/java/com/marklogic/hub/scaffold/Scaffolding.java [hunk 6]
```diff
@@ -29,10 +29,22 @@
...
     static Scaffolding create(String projectDir, DatabaseClient databaseClient) {
         return new ScaffoldingImpl(projectDir, databaseClient);
     }
 
+    /**
+     * Grabs the absolute path of a series of strings
+     * @param first - the minimum required string path to get the absolute path from
+     * @param more - additional paths submitted
+     * @return the absolute path of each string path provided
+     */
```
Comment (paxtonhare) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/scaffold/Scaffolding.java]: this method is only used by the test project. can it be moved out of here?

File: marklogic-data-hub/src/main/java/com/marklogic/hub/scaffold/Scaffolding.java [hunk 7]
```diff
@@ -42,29 +54,96 @@ static String getAbsolutePath(String first, String... more) {
         return absolutePath.toString();
     }
 
+    /**
+     * Returns the directory of the flow
+     * @param entityName - the entity name
+     * @param flowName - the name of the flow
+     * @param flowType - FlowType (sjs or xqy)
+     * @return the directory path as a Path type
+     */
```
Comment (paxtonhare) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/scaffold/Scaffolding.java]: look at where this bad boy is being used. Mostly for internal and testing. Only 1 instance in quickstart that is questionable, at best.

File: marklogic-data-hub/src/main/java/com/marklogic/hub/scaffold/Scaffolding.java [hunk 8]
```diff
@@ -42,29 +54,96 @@ static String getAbsolutePath(String first, String... more) {
...
+     * @throws ScaffoldingValidationException
+     */
     void createRestTransform(String entityName, String transformName,
                              FlowType flowType, CodeFormat codeFormat) throws ScaffoldingValidationException;
 
+    /**
+     * Returns the disk path of the entity
+     * @param entityName - name of the entity as string
+     * @return Path of the entity
+     */
```
Comment (paxtonhare) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/scaffold/Scaffolding.java]: Only used in tests.  maybe move out of public interface..

File: marklogic-data-hub/src/main/java/com/marklogic/hub/scaffold/Scaffolding.java (hunk 7 above)
Comment (aebadirad) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/scaffold/Scaffolding.java]: Going to leave this in due to quickstart until we can figure out a better way to support the quickstart use without invoking a direct implementation.

**Pull request merged**